src/controllers/user.controller.js

```diff
@@ -64,4 +66,37 @@ const logout = async (req, res) => {
   }
 };
 
-export { SignUp, Login, logout };
+const updatePassword = async (req, res, next) => {
+  try {
+    if (req.user.id !== req.params.id) {
+      return res.status(401).json({ error: 'Unauthorised' });
+    }
+    const user = await User.findOne({ where: { id: req.params.id } });
+    const { oldPassword, newPassword } = req.body;
+    // Verify that the old password matches
+    const passwordMatches = await bcrypt.compare(oldPassword, user.password);
+
+    if (!passwordMatches) {
+      return res.status(401).json({ error: 'Incorrect password' });
+    }
+    if (oldPassword === newPassword) {
+      return res
+        .status(401)
+        .json({ error: "New password and old password can't be the same" });
+    }
```

Comment:
this should be handled by a validation middleware

Reply:
Done